### PR TITLE
Add callback for repository doc

### PR DESCRIPTION
**Pull request by cjohnson78**
Example usage:

```
RepositoryDoc.Builder builder = new RepositoryDoc.Builder();

 ...

builder.setCallback(new FutureCallback<GenericJson>() {
    @Override
    public void onSuccess(@Nullable GenericJson result) {
        FileUtils.deleteQuietly(tempFile);
        logger.log(Level.INFO, "Deleting temp file: " + tempFile);
    }

    @Override
    public void onFailure(Throwable t) {
        FileUtils.deleteQuietly(tempFile);
        logger.log(Level.INFO, "Deleting temp file: " + tempFile);
    }
});

 ...

return builder.build();
```

### CODE REVIEW

File: indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/template/RepositoryDoc.java [hunk 1]
```diff
@@ -30,14 +30,12 @@
 import com.google.api.services.cloudsearch.v1.model.RepositoryError;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.Maps;
-import com.google.common.util.concurrent.AsyncFunction;
-import com.google.common.util.concurrent.Futures;
-import com.google.common.util.concurrent.ListenableFuture;
-import com.google.common.util.concurrent.MoreExecutors;
+import com.google.common.util.concurrent.*;
```
Comment (donghanmiao) [inline indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/template/RepositoryDoc.java:33]: Please refraining from using wildcard imports.
https://google.github.io/styleguide/javaguide.html#s3.3.1-wildcard-imports

File: indexing/src/test/java/com/google/enterprise/cloudsearch/sdk/indexing/template/RepositoryDocTest.java [hunk 2]
```diff
@@ -96,6 +100,38 @@ public void testOnlyItem() throws IOException, InterruptedException {
...
 
+  @Test
+  public void testCallback() throws IOException, InterruptedException {
+    Item item = new Item().setName("id1").setAcl(getCustomerAcl());
+    final HashMap callbackResult = new HashMap();
+    RepositoryDoc doc = new RepositoryDoc.Builder()
+            .setItem(item)
+            .setCallback(new FutureCallback<GenericJson>() {
+              @Override
+              public void onSuccess(@Nullable GenericJson result) {
```
Comment (donghanmiao) [inline indexing/src/test/java/com/google/enterprise/cloudsearch/sdk/indexing/template/RepositoryDocTest.java:111]: onSuccess and onFailure are behaving the same, which makes the test less clear. a better way for verifying this is to set a mocked FutureCallback and verify that onSuccess() is called eventually.

**Comment by donghanmiao** [comment]
may you also squash the commits into a single one as you push new changes, it would help maintain a clean commit history of our repo.

**Comment by donghanmiao** [comment]
I previously made a comment to squash the commits, please ignore. I will squash them when during the merge.

**Comment by donghanmiao** [comment]
There was a discussion a while ago between you and Tanmay, where he actually suggested an approach using interface 'CallbackApiOperation' (you should find the thread searching your emails using it), that looks a more generic solution.

File: indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/template/RepositoryDoc.java (hunk 1 above)
Comment (cjohnson78) [reply indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/template/RepositoryDoc.java:33]: Agreed. I had to fix this in my IDE - it was doing it automatically below a certain threshold.

**Comment by cjohnson78** [comment]
Ok, I found that thread with Tanmay. I will test that approach in the connector and see if it achieves the same result.

**Pull request closed without merging**